### PR TITLE
Patch 04/2022

### PR DESCRIPTION
**Pull request by trisyoungs**
A small collection of fixes and improvements in one PR.

### CODE REVIEW

**Comment by rprospero** [comment]
This PR may have a small wait on it.  Homebrew upgraded the ANTLR version on Mac, which seems to be a breaking change, and they have no facility to access old versions.  Until this is fixed, nothing can build on Mac.  I'm currently working on the issue.

File: src/procedure/nodes/node.cpp [hunk 1]
```diff
@@ -300,15 +288,15 @@ bool ProcedureNode::deserialise(LineParser &parser, const CoreData &coreData)
 // Write node data to specified LineParser
 bool ProcedureNode::write(LineParser &parser, std::string_view prefix)
 {
-    // Block Start - does this node have a required name?
-    if (mustBeNamed())
+    // Block Start - node type and name (if specified)
+    if (name_.empty())
```
Comment (IVlaD17) [inline src/procedure/nodes/node.cpp:292]: Shouldn't this be `if(!name_.empty())`?

**Review by IVlaD17 — COMMENTED** [review]
Looks good overall, I can't see anything wrong except for the comment I made.

File: src/procedure/nodes/node.cpp (hunk 1 above)
Comment (trisyoungs) [reply src/procedure/nodes/node.cpp:292]: The code in the `if`/`else` is correct based on this initial condition - I guess the argument here would be to consider the "positive" case first, but I kinda prefer the approach of going for the simpler logic test in the initial condition.

Comment (IVlaD17) [reply src/procedure/nodes/node.cpp:292]: Fair enough.

**Pull request merged**